packages/google_sign_in/lib/google_sign_in.dart

```diff
@@ -217,11 +232,22 @@ class GoogleSignIn {
   /// a Future which resolves to the same user instance.
   ///
   /// Re-authentication can be triggered only after [signOut] or [disconnect].
-  Future<GoogleSignInAccount> signInSilently() {
-    return _addMethodCall('signInSilently').catchError((dynamic _) {
-      // ignore, we promised to be silent.
-      // TODO(goderbauer): revisit when the native side throws less aggressively.
-    });
+  ///
+  /// If [suppressErrors] is set to `false` then returned Future may complete
+  /// with an error.
```

Comment:
We should be able to say a bit more: the error will be a `PlatformException` whose `errorCode` is either `kSignInRequiredError` (when ...), `kSignInCanceledError` (when ...), or `kOtherError` (or some such for the catch-all case) otherwise.